lib/Doctrine/Common/Annotations/AnnotationReader.php

```diff
@@ -174,7 +176,8 @@ public function __construct()
 
         AnnotationRegistry::registerFile(__DIR__ . '/Annotation/IgnoreAnnotation.php');
 
-        $this->parser    = new DocParser;
+        $this->parser = (null === $parser) ? new DocParser() : $parser;
```

Comment:
Given the type-hinting in place, this can be collapsed to `$this->parser = $parser ?: new DocParser()`